lnbits/extensions/boltz/tasks.py

```diff
@@ -0,0 +1,38 @@
+import asyncio
+from loguru import logger
+
+from lnbits.core.models import Payment
+from lnbits.tasks import register_invoice_listener
+
+
+from .crud import get_submarine_swap, get_reverse_submarine_swap, update_swap_status
+
+# from .crud import get_ticket, set_ticket_paid
+
+
+async def wait_for_paid_invoices():
+    invoice_queue = asyncio.Queue()
+    register_invoice_listener(invoice_queue)
+
+    while True:
+        payment = await invoice_queue.get()
+        await on_invoice_paid(payment)
+
+
+async def on_invoice_paid(payment: Payment) -> None:
+    if "boltz" != payment.extra.get("tag"):
+        # not a boltz invoice
+        return
+
+    await payment.set_pending(False)
```

Comment:
Just curious: is this necessary? Some extensions do this, others don't seem to.

Reply:
tbh, i do not know, i did it because everyone is doing it.